### PR TITLE
refactor: Natspec + Splits

### PR DESCRIPTION
**Pull request by maximebrugel**
In preparation for the audit 💅 💅 💅

### CODE REVIEW

File: src/Errors.huff [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+// "VERIFICATION_FAILED"
+#define constant VERIFICATION_FAILED_ERROR = 0x564552494649434154494f4e5f4641494c454400000000000000000000000000
+#define constant VERIFICATION_FAILED_LENGTH = 0x0D
+
+/* -------------------------------------------------------------------------- */
+/*                                   ERRORS                                   */
+/* -------------------------------------------------------------------------- */
+
+/// @notice Reverts with an "JUMP_FAIL" message
```
Comment (obatirou) [inline src/Errors.huff:39]: Natspec is included in huff ?
Maybe dev for errors ?

File: src/HyVM.huff [hunk 2]
```diff
@@ -1,247 +1,189 @@
...
-
-#define macro CONTINUE() = takes(0) returns (1) {
-
-    // load the instruction located at this execution pointer
-    [EXEC_POINTER_MSLOT] mload
-
-    GOTO_AND_ADVANCE() // => [opcode]
+/// @title HyVM
+/// @notice Run arbitrary code on the EVM
+/// @dev This version uses the "no-verifier" abstraction, but "contract-verifier" can be use.
```
Comment (obatirou) [inline src/HyVM.huff:5]:
```suggestion
/// @dev This version uses the "no-verifier" abstraction, but "contract-verifier" can be used.
```

File: src/HyVM.huff [hunk 3]
```diff
@@ -1,247 +1,189 @@
...
 
+#include "Pointer.huff"
```
Comment (obatirou) [inline src/HyVM.huff:7]:
```suggestion
#include "./Pointer.huff"
```

File: src/abstracts/contract-verifier.huff [hunk 4]
```diff
@@ -36,11 +51,33 @@
 #define constant CALLARGS_MSLOT_calldata_len = 0x2E4 // +0x20
 #define constant CALLARGS_MSLOT_calldata_raw = 0x304 // +0x20
 
-// this marks the end of memory space used by the contract verifier
-#define constant HOST_MEMORY_START = 0x460 // +0x40, rounded to upper 20  (= CALLARGS_MSLOT + CALLARGS_TOTALLEN = CALLARGS_MSLOT_calldata_raw + CALLARGS_MAX_DATALEN)
+/// @dev This marks the end of memory space used by the contract verifier.
+//      +0x40, rounded to upper 20 (= CALLARGS_MSLOT + CALLARGS_TOTALLEN = CALLARGS_MSLOT_calldata_raw + CALLARGS_MAX_DATALEN)
```
Comment (obatirou) [inline src/abstracts/contract-verifier.huff:55]: I do not understand this line 
`+0x40, rounded to upper 20`

File: src/abstracts/contract-verifier.huff [hunk 5]
```diff
@@ -83,69 +121,53 @@
...
     // copy sent value to args
     [CALLARGS_MSLOT_value] mstore // => [argOffset, argSize]
 
-
     // store "isTrucated" as false
     0x0 [CALLARGS_MSLOT_isTruncated] mstore
 
-
-    // -> here, we have [argOffset, argSize] on stack
+    // stack => [argOffset, argSize]
```
Comment (obatirou) [inline src/abstracts/contract-verifier.huff:153]: I would put the stack `[argOffset, argSize]` at the end of the line above 
From what I saw and like, putting the state of the stack at the end of a line is nice
Example in huff mate [here](https://github.com/pentagonxyz/huffmate/blob/main/src/tokens/ERC20.huff#L85) or doc [here](https://docs.huff.sh/get-started/huff-by-example/#custom-errors)  
All stack state should be with the same space from code if possible

File: src/abstracts/contract-verifier.huff [hunk 6]
```diff
@@ -83,69 +121,53 @@
...
     swap1 // => [argSize, argOffset]
```
Comment (obatirou) [inline src/abstracts/contract-verifier.huff:154]:
```suggestion
    swap1 // [argSize, argOffset]
```

File: src/Errors.huff [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    [NOT_OWNER_ERROR]
+    [NOT_OWNER_LENGTH]
+    REQUIRE()
+}
+
+/// @notice Reverts with an "CHECK_UNAUTHORIZED" message
+#define macro CHECK_UNAUTHORIZED() = takes (0) returns (0) {
+    [CHECK_UNAUTHORIZED_ERROR]
+    [CHECK_UNAUTHORIZED_LENGTH]
+    REQUIRE()
```
Comment (obatirou) [inline src/Errors.huff:71]: I am not sure about this `REQUIRE()`. 
It takes 3 but only 2 are provided if I am not mistaken     
Same for all errors

File: src/Errors.huff [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+/* -------------------------------------------------------------------------- */
+
+/// @dev Solidity require(...) 
+/// 
+/// MEMORY LAYOUT WHEN THROWN
+/// | sig    || message offset                                                 || message length                                                 || message "revert"                                               |
+/// 0x08c379a  0000000000000000000000000000000000000000000000000000000000000020  0000000000000000000000000000000000000000000000000000000000000006  7265766572740000000000000000000000000000000000000000000000000000
+///
+/// Error `string` MUST be no greater than 32 bytes.
+#define macro REQUIRE() = takes (3) returns (0) {
```
Comment (obatirou) [inline src/Errors.huff:92]: Maybe we could add a test for this one ?
It seems pretty important and from what I saw with errors, it seems there is a mismatch in input provided and what it takes.

File: src/Errors.huff (hunk 1 above)
Comment (maximebrugel) [reply src/Errors.huff:39]: Yes => [Huff Style Guide](https://docs.huff.sh/style-guide/overview/#natspec-comments)

The use of`@notice` is not as obvious as with Solidity. What is "high-level" ? what is "low-level" ?
I prefer to use `@notice` for what a `.huff` file is exposing and `@dev` for internal macros and technical details.

**Comment by obatirou** [comment]
Also, those points could improve readability:
* add takes comments for macro with non zero takes
* fix stack comments: add one when missing +  have same spacing everywhere

**Comment by obatirou** [comment]
One more todo:
Errors/require need to be tested

File: src/abstracts/contract-verifier.huff [hunk 9]
```diff
@@ -36,11 +51,33 @@
...
+    iszero checksstore_success jumpi
 
+    // Check that we're owner (if we are, then we can proceed)
+    ONLY_OWNER()
 
+    checksstore_success:
+}
 
+/// @dev Check that the pending DELEGATECALL operation is OK
 #define macro CHECK_DELEGATECALL() = takes(0) returns (0) {
```
Comment (obatirou) [inline src/abstracts/contract-verifier.huff:81]: It seems it should be `takes(6)`

Comment (obatirou) [reply src/abstracts/contract-verifier.huff:81]: Same for all check of calls

**Comment by obatirou** [comment]
to note: 
* custom errors were remove from the current PR will done in an other PR/issue as it introduces too much changes which need to be tested 
* contract-verifier.huff will be done in an other PR and an issue will be open for further exploration and fix of this feature

**Review by oguimbal — APPROVED** [review]

**Pull request merged**